src/group/libp2p_group_worker.erl

```diff
@@ -509,7 +509,7 @@ kill_pid(undefined) ->
 kill_pid(Pid) ->
     erlang:unlink(Pid),
     erlang:exit(Pid, kill),
-    Pid.
+    undefined.
```

Comment:
This is an unrelated change. When I tried this in https://github.com/helium/erlang-libp2p/pull/238 it failed CT on my local machine, oddly for secure streams, where it previously passed